ciao-4.10/contrib/bin/blanksky

```diff
@@ -137,24 +138,51 @@ def caldb_blanksky(detector):
         return False
 
 
-def t_norm(evtfile,bkgfile,chip,evt_keywords):
+def t_norm(evtfile,bkgfile,chip,evt_keywords,bkg_keywords):
     # use pre-filtered evtfile that matches the filtering applied to the bkgfile
 
     det = evt_keywords["INSTRUME"]
 
-    dmkeypar.punlearn()
-    dmkeypar.infile = bkgfile
-
     if det == "HRC":
-        dmkeypar.keyword = "exposure"
         obs_time = evt_keywords["EXPOSURE"]
+        bkg_time = bkg_keywords["EXPOSURE"]
+    
     else:
-        dmkeypar.keyword = "livtime{}".format(str(chip))
-        obs_time = evt_keywords["LIVTIME{}".format(chip)]
-
-    dmkeypar()
-
-    bkg_time = dmkeypar.rval
+        try:
+            bkg_time = bkg_keywords["LIVETIM{}".format(chip)] 
+        except KeyError:
+            try:
+                bkg_time = bkg_keywords["LIVTIME{}".format(chip)]
+            except KeyError:
+                bkg_time = bkg_keywords["LIVETIME"]
+
+        try:
+            obs_time = evt_keywords["LIVETIM{}".format(chip)] 
+        except KeyError:
+            try:
+                obs_time = evt_keywords["LIVTIME{}".format(chip)]
+            except KeyError:
+                obs_time = evt_keywords["LIVETIME"]
+
+        # if evt_keywords["DATAMODE"] == "GRADED":
```

Comment:
Nick, could we have a comment here saying why the following has been commented out (i.e. why it is worth carrying around)?

Reply:
I've removed the commented out code, as they are unnecessary and just adds confusion.